### PR TITLE
Fixed optional field bug

### PR DESCRIPTION
**Pull request by Larspolo**
## Description
Fixed a problem where a student could not edit optional fields.
Also added some tests to prevent this issue from happening in the future

### CODE REVIEW

File: src/django/VLE/views/entry.py [hunk 1]
```diff
@@ -131,21 +130,31 @@ def partial_update(self, request, *args, **kwargs):
...
+            if old_content.exists():
+                old_content = old_content.first()
+                if old_content.field.pk != field_id:
+                    return response.bad_request('The given content does not match the accompanying field type.')
+                if not data:
+                    old_content.delete()
+                    continue
+
+                entry_utils.patch_entry_content(request.user, entry, old_content, field, data, assignment)
+            # If there was no content before, create new content witht he corrent data.
```
Comment (mjvkeulen) [inline src/django/VLE/views/entry.py]: typos

File: src/django/test/test_entry.py [hunk 2]
```diff
@@ -57,29 +59,65 @@ def test_create(self):
...
+        # Student should be able to edit an optinal field
+        params = {
+            'pk': entry['id'],
+            'content': [{
+                'id': field.pk,
+                'data': 'filled'
+            } for field in fields]
+        }
+        resp = api.update(self, 'entries', params=params.copy(), user=self.student)['entry']
+        print(resp)
```
Comment (mjvkeulen) [inline src/django/test/test_entry.py]: leftover

Comment (engelhamer) [inline src/django/test/test_entry.py]: Leftover print.

File: src/django/VLE/views/entry.py (hunk 1 above)
Comment (mjvkeulen) [inline src/django/VLE/views/entry.py]: with the correct?

Comment (engelhamer) [inline src/django/VLE/views/entry.py]: End of sentence contains some errors.

**Pull request merged**